src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,947 @@
+                # To describe the diagnoser pod which did not reach completed stage
+                arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+                for each_pod in arc_agents_pod_list.items:
+                    pod_name = each_pod.metadata.name
+                    if(pod_name.startswith(job_name)):
+                        # To retrieve the pod logs which is stuck
+                        cmd_get_diagnoser_job_events[4] = "involvedObject.name=" + pod_name
+                        # Using Popen to execute the command and fetching the output
+                        response_kubectl_get_events = Popen(cmd_get_diagnoser_job_events, stdout=PIPE, stderr=PIPE)
+                        output_kubectl_get_events, error_kubectl_get_events = response_kubectl_get_events.communicate()
+                        if response_kubectl_get_events.returncode != 0:
+                            telemetry.set_exception(exception=error_kubectl_get_events.decode("ascii"), fault_type=consts.Kubectl_Get_Events_Failed, summary='Error while doing kubectl get events')
+                            logger.warning("Error while doing kubectl get events")
+                        # Converting output obtained in json format and fetching the clusterconnect-agent feature
+                        events_json = json.loads(output_kubectl_get_events)
+                with open(unfinished_diagnoser_job_path, 'w+') as unfinished_diagnoser_job:
+                    # Adding all the individual events
+                    for events in events_json["items"]:
+                            unfinished_diagnoser_job.write(str(events) + "\n")
+            Popen(cmd_delete_job, stdout=PIPE, stderr=PIPE)
+            diagnoser_output.append("The diagnoser job failed to reach the completed state in the kubernetes cluster.\n")
+            return ""
+
+        else:
+
+            # Fetching the Diagnoser Container logs
+            try:
+                all_pods = corev1_api_instance.list_namespaced_pod('azure-arc')
+                # Traversing thorugh all agents
+                for each_pod in all_pods.items:
+                    # Fethcing the current Pod name and creating a folder with that name inside the timestamp folder
+                    pod_name = each_pod.metadata.name
+                    if(pod_name.startswith(job_name)):
+                        # Creating a text file with the name of the container and adding that containers logs in it
+                        diagnoser_container_log = corev1_api_instance.read_namespaced_pod_log(name=pod_name, container="azure-arc-diagnoser-container", namespace='azure-arc')
+                # Clearing all the resources after fetching the diagnoser container logs
+                Popen(cmd_delete_job, stdout=PIPE, stderr=PIPE)
+            except Exception as e:
+                Popen(cmd_delete_job, stdout=PIPE, stderr=PIPE)
+                return ""
```

Comment:
same comment as above for wherever we're returning empty string

Reply:
I am returning an empty string indicating that no logs were found.